source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -621,46 +691,56 @@ namespace
 
         validateNodesHaveNodedefs(doc);
 
+        // Keep track of renamed nodes:
+        PvtRenamingMapper mapper;
+
         // Load all other elements.
         for (const ElementPtr& elem : doc->getChildren())
         {
             if (!filter || filter(elem))
             {
                 // Make sure the element has not been loaded already.
                 PvtPath path(ROOT_PATH + elem->getName());
-                if (stage->getPrimAtPath(path))
+                if (readOptions && readOptions->conflictResolution == RtReadOptions::SKIP_ELEMENTS && stage->getPrimAtPath(path))
                 {
                     continue;
                 }
 
                 if (elem->isA<Node>())
                 {
-                    readNode(elem->asA<Node>(), stage->getRootPrim(), stage);
+                    readNode(elem->asA<Node>(), stage->getRootPrim(), stage, mapper);
                 }
                 else if (elem->isA<NodeGraph>())
                 {
-                    readNodeGraph(elem->asA<NodeGraph>(), stage->getRootPrim(), stage);
+                    // Always skip if the nodegraph implements a nodedef
```

Comment:
Distinction between the two ways that a nodegraph can be used. One can be renamed, the other is skipped.